File: src/main/java/com/github/zoewithabang/model/BotMessage.java
package com.github.zoewithabang.model;

import java.util.Objects;

public class BotMessage implements IModel
{
    private String name;
    private String message;
    private String description;
    
    public BotMessage(){}
    
    public BotMessage(String name, String message, String description)
    {
        this.name = name;
        this.message = message;
        this.description = description;
    }
    
    public String getName()
    {
        return name;
    }
    
    public void setName(String name)
    {
        this.name = name;
    }
    
    public String getMessage()
    {
        return message;
    }
    
    public void setMessage(String message)
    {
        this.message = message;
    }
    
    public String getDescription()
    {
        return description;
    }
    
    public void setDescription(String description)
    {
        this.description = description;
    }
    
    @Override
    public boolean equals(Object o)
    {
        if(this == o) return true;
        if(o == null || getClass() != o.getClass()) return false;
        BotMessage that = (BotMessage) o;
        return Objects.equals(name, that.name) &&
            Objects.equals(message, that.message) &&
            Objects.equals(description, that.description);
    }
    
    @Override
    public int hashCode()
    {
        return Objects.hash(name, message, description);
    }
    
    @Override
    public String toString()
    {
        return "BotMessage{" +
            "name='" + name + '\'' +
            ", message='" + message + '\'' +
            ", description='" + description + '\'' +
            '}';
    }
}

File: src/main/kotlin/com/github/zoewithabang/model/BotMessage.kt
package com.github.zoewithabang.model

class BotMessage {
    var name: String? = null
    var message: String? = null
    var description: String? = null

    constructor()

    constructor(name: String, message: String, description: String) {
        this.name = name
        this.message = message
        this.description = description
    }

    override fun toString(): String {
        return "BotMessage(name=$name, message=$message, description=$description)"
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as BotMessage

        if (name != other.name) return false
        if (message != other.message) return false
        if (description != other.description) return false

        return true
    }

    override fun hashCode(): Int {
        var result = name?.hashCode() ?: 0
        result = 31 * result + (message?.hashCode() ?: 0)
        result = 31 * result + (description?.hashCode() ?: 0)
        return result
    }
}
